contrib/native/client/src/clientlib/drillClientImpl.cpp

```diff
@@ -678,7 +678,10 @@ connectionStatus_t DrillClientImpl::handleAuthentication(const DrillUserProperti
         m_io_service.reset();
         return CONN_SUCCESS;
     } else {
-        logMsg << m_encryptionCtxt << ", Error: " << m_saslResultCode;
+        logMsg << m_encryptionCtxt
+               << ", Mechanism: " << m_saslAuthenticator->getAuthMechanismName()
+               << ", Error: " << m_saslResultCode
+               << ", Cause: " << m_saslAuthenticator->getErrorMessage(m_saslResultCode);
```

Comment:
How about adding this cause message string in the [if condition](https://github.com/apache/drill/pull/992/files#diff-8e6df071d8ca863fcfa578892944c1dcR661) too ?